lib/phoenix/channel.ex

```diff
@@ -279,16 +293,19 @@ defmodule Phoenix.Channel do
               {:shutdown, :left | :closed} |
               term
 
-  defmacro __using__(_) do
+  defmacro __using__(opts \\ []) do
     quote do
       @behaviour unquote(__MODULE__)
       @on_definition unquote(__MODULE__)
       @before_compile unquote(__MODULE__)
       @phoenix_intercepts []
+      @log unquote(opts[:log] || [join: :info, handle_in: :info])
```

Comment:
Actually, I am not a big fan of this API. What does an empty list mean? If I configure only one of them, like `[join: :info]`, does it mean I don't  want to log the other?

Also controllers pass the log_level explicitly as the instrumentation metadata. Is there any chance we could do something similar here?